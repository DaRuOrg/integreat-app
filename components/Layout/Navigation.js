import React from 'react'

class Navigation extends React.Component {
  componentDidMount () {

  }

  componentWillUnmount () {

  }

  render () {
    return (
      <nav>

      </nav>
    )
  }
}

export default Navigation
