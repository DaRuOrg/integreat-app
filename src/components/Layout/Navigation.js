export default class Navigation {
  constructor (home, location, language) {
    this._home = home
    this._location = location
    this._language = language
  }

  get home () {
    return this._home
  }

  get location () {
    return this._location
  }

  get language () {
    return this._language
  }
}
