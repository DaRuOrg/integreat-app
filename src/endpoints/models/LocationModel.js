import { isEmpty } from 'lodash/lang'

export default class LocationModel {
  constructor (name, path, live) {
    this._code = name
    this._name = path
    this._live = live
  }

  get live () {
    return this._live
  }

  get name () {
    return this._code
  }

  get path () {
    return this._name
  }

  get category () {
    return isEmpty(this._code) ? '?' : this._code[0].toUpperCase()
  }
}
