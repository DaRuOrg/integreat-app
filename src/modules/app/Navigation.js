class Navigation {
  constructor (location, language) {
    this._location = location
    this._language = language
  }

  get categories () {
    if (!this._location) {
    }
    return `/${this._location}/${this._language}`
  }

  get locationSelector () {
    if (!this._language) {
      return '/'
    }
    return `/${this._language}`
  }

  get search () {
    if (!this._language || !this._location) {
      return '/'
    }
    return `/${this._location}/${this._language}/search`
  }

  get disclaimer () {
    if (!this._language || !this._location) {
      return '/disclaimer'
    }
    return `/${this._location}/${this._language}/disclaimer`
  }

  get events () {
    if (!this._language || !this._location) {
      return '/'
    }
    return `/${this._location}/${this._language}/events`
  }

  get extras () {
    if (!this._language || !this._location) {
      return '/'
    }
    return `/${this._location}/${this._language}/extras`
  }
}

export default Navigation
