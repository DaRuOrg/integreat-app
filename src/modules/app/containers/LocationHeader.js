import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'
import { connect } from 'react-redux'
import compose from 'lodash/fp/compose'

import Navigation from 'modules/app/Navigation'
import LanguageSelector from './LanguageSelector'
import searchIcon from '../assets/magnifier.svg'
import locationIcon from '../assets/location-icon.svg'
import languageIcon from '../assets/language-icon.svg'
import LocationModel from 'modules/endpoint/models/LocationModel'
import withFetcher from 'modules/endpoint/hocs/withFetcher'
import LOCATIONS_ENDPOINT from 'modules/endpoint/endpoints/locations'
import Header from 'modules/layout/components/Header'

class LocationHeader extends React.Component {
  static propTypes = {
    navigation: PropTypes.instanceOf(Navigation).isRequired,
    location: PropTypes.instanceOf(LocationModel).isRequired,
    route: PropTypes.string.isRequired
  }

  isCategoriesEnabled = () => this.isExtrasEnabled() || this.isEventsEnabled()

  isExtrasEnabled = () => this.props.location.extrasEnabled

  isEventsEnabled = () => this.props.location.eventsEnabled

  isExtrasSelected = () => this.props.route === '/:location/:language/extras'

  isCategoriesSelected = () => ['/:location/:language', '/:location/:language/*'].includes(this.props.route)

  isEventsSelected = () => this.props.route === '/:location/:language/events(/:id)'

  getActionItems () {
    const {navigation} = this.props
    return [
      {href: navigation.search, iconSrc: searchIcon},
      {href: navigation.locationSelector, iconSrc: locationIcon},
      {dropDownNode: <LanguageSelector />, iconSrc: languageIcon}
    ]
  }

  getMenuItems () {
    const {t, navigation} = this.props
    const extras = this.isExtrasEnabled() &&
      {href: navigation.extras, active: this.isExtrasSelected(), text: t('extras')}

    const categories = this.isCategoriesEnabled() &&
      {href: navigation.categories, active: this.isCategoriesSelected(), text: t('categories')}

    const events = this.isEventsEnabled() &&
      {href: navigation.events, active: this.isEventsSelected(), text: t('news')}

    return [extras, categories, events].filter(item => item)
  }

  render () {
    return <Header actionItems={this.getActionItems()} menuItems={this.getMenuItems()} />
  }
}

const mapStateToProps = (state) => ({
  route: state.router.route,
  navigation: new Navigation(state.router.params.location, state.router.params.language)
})

export default compose(
  connect(mapStateToProps),
  withFetcher(LOCATIONS_ENDPOINT, true, true),
  translate('app')
)(LocationHeader)
