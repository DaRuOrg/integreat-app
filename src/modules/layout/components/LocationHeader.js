import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'

import LanguageSelector from '../../common/containers/LanguageSelector'
import searchIcon from '../assets/magnifier.svg'
import locationIcon from '../assets/location-icon.svg'
import languageIcon from '../assets/language-icon.svg'
import LocationModel from 'modules/endpoint/models/LocationModel'
import Header from 'modules/layout/components/Header'
import HeaderNavigationItem from '../HeaderNavigationItem'
import HeaderActionItem from '../HeaderActionItem'
import SearchPage from '../../../routes/search/containers/SearchPage'
import LandingPage from '../../../routes/landing/containers/LandingPage'
import CategoriesPage from '../../../routes/categories/containers/CategoriesPage'
import EventsPage from '../../../routes/events/containers/EventsPage'

class LocationHeader extends React.Component {
  static propTypes = {
    matchRoute: PropTypes.func.isRequired,
    currentParams: PropTypes.object.isRequired,
    location: PropTypes.instanceOf(LocationModel).isRequired,
    route: PropTypes.string.isRequired,
    t: PropTypes.func.isRequired
  }

  getActionItems () {
    const {matchRoute, currentParams} = this.props
    return [
      new HeaderActionItem({href: matchRoute(SearchPage).stringify(currentParams), iconSrc: searchIcon}),
      new HeaderActionItem({href: matchRoute(LandingPage).stringify(currentParams), iconSrc: locationIcon}),
      new HeaderActionItem({dropDownNode: <LanguageSelector />, iconSrc: languageIcon})
    ]
  }

  getNavigationItems () {
    const {t, matchRoute, currentParams, route} = this.props

    const isEventsEnabled = () => this.props.location.eventsEnabled
    const isExtrasEnabled = () => this.props.location.extrasEnabled
    const isCategoriesEnabled = () => isExtrasEnabled() || isEventsEnabled()

    const isExtrasSelected = () => false // todo for WEBAPP-64: test and verify this
    const isCategoriesSelected = () => matchRoute(CategoriesPage).hasPath(route)
    const isEventsSelected = () => matchRoute(EventsPage).hasPath(route)

    const extras = isExtrasEnabled() &&
      new HeaderNavigationItem({
        href: '/',
        active: isExtrasSelected(),
        text: t('extras')
      })

    const categories = isCategoriesEnabled() &&
      new HeaderNavigationItem({
        href: matchRoute(CategoriesPage).stringify(currentParams),
        active: isCategoriesSelected(),
        text: t('categories')
      })

    const events = isEventsEnabled() &&
      new HeaderNavigationItem({
        href: matchRoute(EventsPage).stringify(currentParams),
        active: isEventsSelected(),
        text: t('news')
      })

    return [extras, categories, events].filter(isEnabled => isEnabled)
  }

  render () {
    return <Header actionItems={this.getActionItems()} navigationItems={this.getNavigationItems()} />
  }
}

export default translate('app')(LocationHeader)
