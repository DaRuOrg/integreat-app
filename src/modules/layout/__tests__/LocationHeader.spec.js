import { shallow } from 'enzyme'
import React from 'react'
import Navigation from '../../app/Navigation'
import LocationHeader from '../components/LocationHeader'
import LocationModel from 'modules/endpoint/models/LocationModel'

describe('LocationHeader', () => {
  const navigation = new Navigation('location1', 'language1')
  const createLocation = (extrasEnabled, eventsEnabled) => new LocationModel({
    name: 'Mambo Nr. 5',
    code: 'location1',
    eventsEnabled,
    extrasEnabled
  })

  describe('MenuItems', () => {
    test('should be empty, if extras and news are both disabled', () => {
      const component = shallow(<LocationHeader navigation={navigation} location={createLocation(false, false)}
                                                route='' />)
      expect(component.dive().prop('menuItems')).toMatchSnapshot()
    })

    test('should show categories, if extras or news are enabled', () => {
      const extrasComp = shallow(<LocationHeader navigation={navigation} location={createLocation(true, false)}
                                                 route='' />)
      const eventsComp = shallow(<LocationHeader navigation={navigation} location={createLocation(false, true)}
                                                 route='' />)

      expect(extrasComp.dive().prop('menuItems')).toMatchSnapshot()
      expect(eventsComp.dive().prop('menuItems')).toMatchSnapshot()
    })

    test('should show extras, categories, events in this order', () => {
      const component = shallow(<LocationHeader navigation={navigation} location={createLocation(true, true)}
                                                route='' />)
      expect(component.dive().prop('menuItems')).toMatchSnapshot()
    })

    test('should highlight categories if route corresponds', () => {
      const route1Component = shallow(<LocationHeader navigation={navigation} location={createLocation(true, true)}
                                                      route='/:location/:language' />)
      const route2Component = shallow(<LocationHeader navigation={navigation} location={createLocation(true, true)}
                                                      route='/:location/:language/*' />)
      expect(route1Component.dive().prop('menuItems')).toMatchSnapshot()
      expect(route2Component.dive().prop('menuItems')).toEqual(route1Component.dive().prop('menuItems'))
    })

    test('should highlight events if route corresponds', () => {
      const component = shallow(<LocationHeader navigation={navigation} location={createLocation(true, true)}
                                                route='/:location/:language/events(/:id)' />)

      expect(component.dive().prop('menuItems')).toMatchSnapshot()
    })

    // todo for WEBAPP-64: should highlight extras if route corresponds
  })

  test('should match snapshot', () => {
    const component = shallow(<LocationHeader navigation={navigation} location={createLocation(true, true)} route='' />)
    expect(component.dive()).toMatchSnapshot()
  })
})
