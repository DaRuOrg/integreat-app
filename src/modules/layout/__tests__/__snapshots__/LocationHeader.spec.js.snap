// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationHeader NavigationItems should be empty, if extras and news are both disabled 1`] = `Array []`;

exports[`LocationHeader NavigationItems should highlight categories if route corresponds 1`] = `
Array [
  Object {
    "active": false,
    "href": "/location1/language1/extras",
    "text": "app:extras",
  },
  Object {
    "active": true,
    "href": "/location1/language1",
    "text": "app:categories",
  },
  Object {
    "active": false,
    "href": "/location1/language1/events",
    "text": "app:news",
  },
]
`;

exports[`LocationHeader NavigationItems should highlight events if route corresponds 1`] = `
Array [
  Object {
    "active": false,
    "href": "/location1/language1/extras",
    "text": "app:extras",
  },
  Object {
    "active": false,
    "href": "/location1/language1",
    "text": "app:categories",
  },
  Object {
    "active": true,
    "href": "/location1/language1/events",
    "text": "app:news",
  },
]
`;

exports[`LocationHeader NavigationItems should show categories, if extras or news are enabled 1`] = `
Array [
  Object {
    "active": false,
    "href": "/location1/language1/extras",
    "text": "app:extras",
  },
  Object {
    "active": false,
    "href": "/location1/language1",
    "text": "app:categories",
  },
]
`;

exports[`LocationHeader NavigationItems should show categories, if extras or news are enabled 2`] = `
Array [
  Object {
    "active": false,
    "href": "/location1/language1",
    "text": "app:categories",
  },
  Object {
    "active": false,
    "href": "/location1/language1/events",
    "text": "app:news",
  },
]
`;

exports[`LocationHeader NavigationItems should show extras, categories, events in this order 1`] = `
Array [
  Object {
    "active": false,
    "href": "/location1/language1/extras",
    "text": "app:extras",
  },
  Object {
    "active": false,
    "href": "/location1/language1",
    "text": "app:categories",
  },
  Object {
    "active": false,
    "href": "/location1/language1/events",
    "text": "app:news",
  },
]
`;

exports[`LocationHeader should match snapshot 1`] = `
<Header
  actionItems={
    Array [
      Object {
        "href": "/location1/language1/search",
        "iconSrc": "test-file-stub",
      },
      Object {
        "href": "/language1",
        "iconSrc": "test-file-stub",
      },
      Object {
        "dropDownNode": <Connect(Connect(languagesFetcher)) />,
        "iconSrc": "test-file-stub",
      },
    ]
  }
  navigationItems={
    Array [
      Object {
        "active": false,
        "href": "/location1/language1/extras",
        "text": "app:extras",
      },
      Object {
        "active": false,
        "href": "/location1/language1",
        "text": "app:categories",
      },
      Object {
        "active": false,
        "href": "/location1/language1/events",
        "text": "app:news",
      },
    ]
  }
/>
`;
